src/featurelist.h

```diff
@@ -7,52 +7,202 @@
+            FEATURE_NOTIFICATION_CONTROL,       // ID
+            L18nStrings::FeatureNamesNotificationControl,
+            false,                  // Is Major Feature
+            L18nStrings::Empty,     // Description
+            L18nStrings::Empty,     // LongDescr
+            "",                     // ImagePath
+            "",                     // IconPath
+            true,                   // released
+            true,                   // Can be enabled in devmode
+            true                    // Check OS
+        );
+    const Feature* CustomDNS = new Feature(
+            FEATURE_CUSTOM_DNS,     // ID
+            L18nStrings::FeatureNamesCustomDNS,
+            true,                   // Is Major Feature
+            L18nStrings::Empty,     // Description
+            L18nStrings::Empty,     // LongDescr
+            "",                     // ImagePath
+            "",                     // IconPath
+            true,                   // released
+            true                    // Can be enabled in devmode
+        );
+    const Feature* InAppAuth = new Feature(
+            FEATURE_IN_APP_AUTH,    // ID
+            L18nStrings::FeatureNamesInAppAuth,
+            false,                  // Is Major Feature
+            L18nStrings::Empty,     // Description
+            L18nStrings::Empty,     // LongDescr
+            "",                     // ImagePath
+            "",                     // IconPath
+            false,                  // released
+            true,                   // Can be enabled in devmode
+            false                   // Check OS
+        );
+    const Feature* InAppAccountCreate = new Feature(
+            FEATURE_IN_APP_ACCOUNT_CREATE,  // ID
+            L18nStrings::FeatureNamesInAppAccountCreate,
+            true,                           // Is Major Feature
+            L18nStrings::Empty,             // Description
+            L18nStrings::Empty,             // LongDescr
+            "",                             // ImagePath
+            "",                             // IconPath
+            true,                           //released
+            false,                          // Can be enabled in devmode
+            true                            // Check OS
+        );
+    const Feature* InAppPurchase = new Feature(
+            FEATURE_IN_APP_PURCHASE,// ID
+            L18nStrings::FeatureNamesInAppPurchase,
+            true,                   // Is Major Feature
+            L18nStrings::Empty,     // Description
+            L18nStrings::Empty,     // LongDescr
+            "",                     // ImagePath
+            "",                     // IconPath
+            true,          // released
+            false,         // Can be enabled in devmode
+            true           // Check OS
+        );
+
+    QList<const Feature*> getFeatureList() const{
+        return m_featurelist;
+    }
+
+private:
+    QList<const Feature*> m_featurelist{
```

Comment:
This makes it quite complex to "add" new features. We must add an ID, a new Feature object, and add it to this list.

Can we do something similar to how the tests are registered in the unit-test? Something like:

1. in featurelist.cpp:
```
namespace {
QHash<QString, const Feature*> s_features;
}
```
2. in the CTOR of Feature:
```
Feature::Feature(const QString& id) {
  s_featureList.push(id, this);
 }
```
3. for QML we can expose things using QQmlPropertyMap (maybe)
4. For C++, we need a getter method based on strings. and to avoid errors, if the string doesn't match an existing feature, we crash the client.